pkg/endpoints/cluster.go

```diff
@@ -162,3 +164,70 @@ func (r Resource) GetProperties(request *restful.Request, response *restful.Resp
 	properties := Properties{InstallNamespace: os.Getenv("INSTALLED_NAMESPACE")}
 	response.WriteEntity(properties)
 }
+
+// Get dashboard version
+func (r Resource) GetDashboardVersion(request *restful.Request, response *restful.Response) {
+	config, err := rest.InClusterConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	properties := Properties{InstallNamespace: os.Getenv("INSTALLED_NAMESPACE")}
+	version := ""
+
+	api1 := clientset.AppsV1()
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=tekton-dashboard",
+	}
+	deployments, err := api1.Deployments(properties.InstallNamespace).List(listOptions)
+
+	for _, deployment := range deployments.Items {
+		deploymentLabels := deployment.GetLabels()
+		version = deploymentLabels["version"]
+	}
+
+	if version == "" {
+		response.WriteEntity("Unknown")
+		return
+	}
+	response.WriteEntity(version)
+	return
+}
+
+// Get pipelines version
+func (r Resource) GetPipelineVersion(request *restful.Request, response *restful.Response) {
+	config, err := rest.InClusterConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	version := ""
+
+	api1 := clientset.AppsV1()
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app.kubernetes.io/component=controller,app.kubernetes.io/name=tekton-pipelines",
+	}
+	deployments, err := api1.Deployments("tekton-pipelines").List(listOptions)
+
+	for _, deployment := range deployments.Items {
+		deploymentAnnotations := deployment.Spec.Template.GetAnnotations()
+		version = deploymentAnnotations["tekton.dev/release"]
+
+		if version == "" {
+			deploymentImage := deployment.Spec.Template.Spec.Containers[0].Image
+			if strings.Contains(deploymentImage, "pipeline/cmd/controller") {
+				s := strings.SplitAfter(deploymentImage, ":")
```

Comment:
Make sure these are safe/error checked for $(go panic reduction chance--)

Reply:
I have added some safe / error checking to make sure that it doesnt break or return the wrong version back